CMakeLists.txt

```diff
@@ -38,14 +40,51 @@ enable_testing()
 
 list(APPEND CMAKE_MODULE_PATH ${PROJECT_SOURCE_DIR}/cmake)
 
+# There are cyclical dependencies ( betwen libopencensus_trace.so and
+# libopencensus_context.so ) that prevent from building the code as is as shared
+# libraries without undefined symbols...
+#
+# Moreover, because of the way I built gRPC c++ apis within google-cloud-cpp as
+# shared libs, with a specific abseil version _as shared lib_
+
+if(BUILD_SHARED_LIBS)
+  set(POSITION_INDEPENDENT_CODE ON)
+  set(BUILD_SHARED_LIBS NO)
+endif()
+# clearly there are problems with cyclical deps in this code ( context, trace )
+if(BUILD_SHARED_LIBS)
+  add_link_options("LINKER:--no-undefined")
```

Comment:
this requires a version of cmake that's more recent than the one in the CI containers... consider a platform-aware version of this:
```
set( CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,--no-undefined" )
```